src/createStoreController.ts

```diff
@@ -41,42 +45,78 @@ export default async function (
 }> {
   const store = await storePath(opts.prefix, opts.store)
   const connectionInfoDir = serverConnectionInfoDir(store)
-  try {
-    const serverJson = await loadJsonFile(path.join(connectionInfoDir, 'server.json'))
+  const serverJsonPath = path.join(connectionInfoDir, 'server.json')
+  let serverJson = await tryLoadServerJson(serverJsonPath, false)
+  if (serverJson !== null) {
     logger.info('A store server is running. All store manipulations are delegated to it.')
     return {
       ctrl: await connectStoreController(serverJson.connectionOptions), // tslint:disable-line
       path: store,
     }
-  } catch (err) {
-    if (err.code !== 'ENOENT') throw err
   }
   if (opts.useStoreServer) {
     runServerInBackground(store)
-    const operation = retry.operation()
-
-    return new Promise<{
-      ctrl: StoreController,
-      path: string,
-    }>((resolve, reject) => {
-      operation.attempt(async (currentAttempt) => {
-        try {
-          const serverJson = await loadJsonFile(path.join(connectionInfoDir, 'server.json'))
-          logger.info('A store server has been started. To stop it, use \`pnpm server stop\`')
-          resolve({
-            ctrl: await connectStoreController(serverJson.connectionOptions), // tslint:disable-line
-            path: store,
-          })
-          return
-        } catch (err) {
-          if (!operation.retry(err)) {
-            reject(operation.mainError())
-          }
-        }
-      })
-    })
+    serverJson = await tryLoadServerJson(serverJsonPath, true)
+    logger.info('A store server has been started. To stop it, use \`pnpm server stop\`')
+    return {
+      ctrl: await connectStoreController(serverJson.connectionOptions), // tslint:disable-line
+      path: store,
+    }
   }
   return await createStore(Object.assign(opts, {
     store,
   }))
 }
+
+export async function tryLoadServerJson (serverJsonPath: string, shouldRetryOnNoent: boolean) {
```

Comment:
use an `options` object to set `{shouldRetryOnNoent: true}` because otherwise it is unclear what the bool means when tryLoadServerJson is called